### DISCUSSION TITLE
Diagrama de clase

### DISCUSSION THREAD

**Discussion by Rosmira** [SarayGal/Unac-Virtual-Tracking#75] [Libro de Conocimiento]
**DIFINICION DE CLASE**

Es una colección de objetos similares. Estos objetos deben tener los mismos atributos con valores posiblemente diferentes asignados a estos, y el mismo conjunto de métodos que definen su comportamiento.

**ATRIBUTOS Y METODOS**
Los atributos son valores almacenados por los objetos de una clase.​

Los métodos son secuencias de pasos para determinar un comportamiento. ​

Los métodos pueden ser llamados (invocados) en principio, solamente desde dentro de la clase o a través de una instancia u objeto de la misma.​

![image](https://user-images.githubusercontent.com/26470460/115329561-15cecb00-a158-11eb-8ec9-d15c24710b03.png)

**Relaciones entre clases​**
Es posible establecer relaciones entre dos clases por varias razones. ​

Una relación es una conexión entre elementos y entre las más importantes tenemos las dependencias, las asociaciones y las generalizaciones; gráficamente, una relación se puede representar por una línea, usándose diferentes tipos de línea para diferenciar los tipos de relaciones. ​
![image](https://user-images.githubusercontent.com/26470460/115329995-d05ecd80-a158-11eb-8df7-40b9f2f37a2c.png)
![image](https://user-images.githubusercontent.com/26470460/115330049-f5534080-a158-11eb-8a24-0c4d42546750.png)
![image](https://user-images.githubusercontent.com/26470460/115330136-1c117700-a159-11eb-985b-cbbe6dae4c52.png)

**Comment by Rosmira** [comment]
Autor Jose Baldovino

Para la creación de diagramas se utilizó la plataforma LuciChard.
Aprendí como hacer diagramas de una manera fácil, a continuación, algunos pasos para la creación de estos diagramas:

Abre un documento en blanco o empieza con una plantilla.
Activa la biblioteca de figuras UML. En el lado izquierdo del editor de Lucidchart, haz clic en "Figuras". Una vez que estés en el Administrador de bibliotecas de figuras, marca "UML" y haz clic en "Guardar".
Desde las bibliotecas que recién agregaste, selecciona la figura que quieras y arrástrala de la caja de herramientas al lienzo.
Modela el flujo de procesos trazando líneas entre las figuras mientras agregas texto.
El diagrama de clases está compuesto por tres partes:
Sección superior: Contiene el nombre de la clase. Esta sección siempre es necesaria, ya sea que estés hablando del clasificador o de un objeto.
Sección central: Contiene los atributos de la clase. Usa esta sección para describir cualidades de la clase. Esto solo es necesario al describir una instancia específica de una clase.
Sección inferior: Incluye operaciones de clases (métodos). Esto está organizado en un formato de lista. Cada operación requiere su propia línea. Las operaciones describen cómo una clase puede interactuar con los datos.
En cuanto a los modificadores de acceso a miembros encontré que todas las clases poseen diferentes niveles de acceso en función del modificador de acceso (visibilidad). A continuación, los diferentes niveles de acceso con sus símbolos correspondientes:
Público (+)
Privado (-)
Protegido (#)
Paquete (~)
Derivado (/)
Estático (subrayado)

**Comment by JoseBaldovino1014** [comment]
DIAGRAMA DE CLASE
![Diagrama de Clases - Clase UML (1)](https://user-images.githubusercontent.com/32313570/115549781-a7306100-a26e-11eb-9846-fdcf47c16525.jpeg)

**Comment by Rosmira** [comment]
Ejemplo Diagrama de clase
![image](https://user-images.githubusercontent.com/26470460/115331483-980cbe80-a15b-11eb-9f5a-2a3b759da9a5.png)
